pkg/apis/migration/migration.go

```diff
@@ -407,3 +415,129 @@ func (volumeMigration *volumeMigration) registerVolume(ctx context.Context, volu
 	}
 	return volumeID.Id, nil
 }
+
+// cleanupStaleCRDInstances helps in cleaning up stale volume migration CRD instances
+func (volumeMigration *volumeMigration) cleanupStaleCRDInstances(ctx context.Context) {
+	log := logger.GetLogger(ctx)
+	config, err := k8s.GetKubeConfig(ctx)
+	if err != nil {
+		log.Errorf("failed to get kubeconfig. err: %v", err)
+		return
+	}
+	crdGroupClient, err := k8s.NewClientForGroup(ctx, config, CRDGroupName)
+	if err != nil {
+		log.Errorf("Failed to create new client for group %q . Err: %+v", CRDGroupName, err)
+		return
+	}
+	// Clean up routine to cleanup successful CnsRegisterVolume instances
+	err = volumeMigrationInstance.watcher(ctx)
+	if err != nil {
+		log.Error("Failed to watch on config file for changes to VolumeMigrationCRCleanupIntervalInHours. Error: %+v", err)
+		return
+	}
+	go func() {
+		for {
```

Comment:
instead of this kind of for look can use `for _ = range ticker.C {`

https://tutorialedge.net/golang/go-ticker-tutorial/

Reply:
Done